sansa-rdf-spark/src/main/scala/net/sansa_stack/rdf/spark/model/df/TripleOps.scala

```diff
@@ -0,0 +1,309 @@
+    triples.rdd.map(row =>
+      Triple.create(
+        NodeFactory.createURI(row.getString(0)),
+        NodeFactory.createURI(row.getString(1)),
+        //NodeFactory.createLiteral(row.getString(2))
+        if (row.getString(2).startsWith("http:")) NodeFactory.createURI(row.getString(2)) else NodeFactory.createLiteral(row.getString(2))))
+  }
+
+  /**
+   * Convert a DataFrame of Triple into a Dataset of Triple.
+   *
+   * @param triples DataFrame of triples.
+   * @return a Dataset of triples.
+   */
+  def toDS(triples: DataFrame): Dataset[Triple] = {
+    val spark: SparkSession = SparkSession.builder().getOrCreate()
+    implicit val encoder = Encoders.kryo[Triple]
+    //triples.as[Triple]
+    spark.createDataset[Triple](toRDD(triples))
+  }
+
+  /**
+   * Get triples.
+   *
+   * @param triples DataFrame of triples.
+   * @return DataFrame which contains list of the triples.
+   */
+  def getTriples(triples: DataFrame): DataFrame =
+    triples
+
+  /**
+   * Get subjects.
+   *
+   * @param triples DataFrame of triples.
+   * @return DataFrame which contains list of the subjects.
+   */
+  def getSubjects(triples: DataFrame): DataFrame =
+    triples.select("s")
+
+  /**
+   * Get predicates.
+   *
+   * @param triples DataFrame of triples.
+   * @return DataFrame which contains list of the predicates.
+   */
+  def getPredicates(triples: DataFrame): DataFrame =
+    triples.select("p")
+
+  /**
+   * Get objects.
+   *
+   * @param triples DataFrame of triples.
+   * @return DataFrame which contains list of the objects.
+   */
+  def getObjects(triples: DataFrame): DataFrame =
+    triples.select("o")
+
+  /**
+   * Returns an DataFrame of triples that match with the given input.
+   *
+   * @param triples DataFrame of triples
+   * @param subject the subject
+   * @param predicate the predicate
+   * @param object the object
+   * @return DataFrame of triples
+   */
+  def find(triples: DataFrame, subject: Option[String] = None, predicate: Option[String] = None, `object`: Option[String] = None): DataFrame = {
+
+    var sql = s"SELECT s, p, o FROM TRIPLES"
+
+    if (subject.isDefined || predicate.isDefined || `object`.isDefined) {
+      sql += " WHERE "
+
+      val conditions = scala.collection.mutable.ListBuffer[String]()
+
+      if (subject.isDefined) conditions += s"s = '${subject.get}'"
+      if (predicate.isDefined) conditions += s"p = '${predicate.get}'"
+      if (`object`.isDefined) conditions += s"o = '${`object`.get}'"
+
+      sql += conditions.mkString(" AND ")
+    }
+    println(sql)
+    println(subject.get + ", " + predicate.get + ", " + `object`.get)
+    triples.sqlContext.sql(sql)
+  }
+
+  /**
+   * Returns an DataFrame of triples that match with the given input.
+   *
+   * @param triples DataFrame of triples
+   * @param triple the triple to be checked
+   * @return DataFrame of triples that match the given input
+   */
+  def find(triples: DataFrame, triple: Triple): DataFrame = {
```

Comment:
`triple` argument is a triple pattern, please rename
Also I would rename find to query, because essentially it is triple pattern matching against triples Dataframe

Would it be more understandable to call `triples` argument `graphDF`?

Reply:
`triple` is supposed to be a specific instance represented as Triple of (s, p, o). It is not meant to be a triple-pattern as BGPs. So why we are naming it as `find` instead of `query` it is because it just looks/ searches for a specific instance instead of SPARQL query. 
and we name `triples` just to be consistent all over the representations (RDD/DataFrame/Dataset). 
But, I guess we can refactor it if needed. Let me know your opinion.

Reply:
As far as I can see, most of the code is from what I used in Inference layer during evaluation of the different datastructures - that being said, I did it already with `Dataset`, `DataFrame` and `RDD` but didn't come up to a final "best" solution.
Indeed, for abstraction and to improve the meaning, one could also use [`TriplePattern`](https://jena.apache.org/documentation/javadoc/jena/org/apache/jena/reasoner/TriplePattern.html) class of Apache Jena - it wraps a `Triple` but has a stronger meaning.

Reply:
Indeed, it is.

As we have discussed previously we have to align Inference layer with RDF layer where we use the same reader/ops from SANSA-RDF to all top layers of SANSA. This was a first step towards aligning these two layers even better. If you can re-use these wrappers for Inference layers would be great to use/expand this code here and move some parts from Inference layer to RDF layer as well. 

Please, feel free to expand/remove some code from here and then we can easily use this as an input for Inference layer as well.  

As you know it already, the idea is to have something like this for Inference layer as well : 

```scala
val triples = spark.rdf(lang)(path)
val inferedGraph = triples.inference(profile) //which should besically return an infered graph of RDD[Triple], in case we use RDDs.
```
Let me know what do you think about this.